### ISSUE TITLE
Miner never schedules any work on other miners, since 0.7.0 only.

### ISSUE THREAD

**Issue by Reisen** [filecoin-project/lotus#3794]
**Describe the problem**

My miner is observing the presense of workers, but is not assigning work to them. It has been in this state ever since 0.7.0, coming up to a couple of days now. In particular, scheduling logs appear to ignore its own queue. The full state of my machine for context can be seen here: [machine overview](https://paste.rs/pva)

Here are the states that make me think this is a genuine bug:
- Miners are correctly observed.
- All miners have sufficient disk space (2.5TiB each NVME free)
- Miners sit in "Waiting for tasks" state indefinitely.
- When scheduling logs are presented, 6 windows are seen, and yet it will still attempt to assign a 3rd job to an overloaded worker (0, the only worker receiving work).

Here is an example log where the main miner + 1 worker were running, which can be seen to lead to the assignment state in the above paste:

```
2020-09-12T13:49:00.379Z        DEBUG   advmgr  sector-storage/sched.go:324     SCHED 1 queued; 4 open windows
2020-09-12T13:49:00.382Z        DEBUG   advmgr  sector-storage/sched.go:415     SCHED windows: [{allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>}
 todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]}]
2020-09-12T13:49:00.382Z        DEBUG   advmgr  sector-storage/sched.go:416     SCHED Acceptable win: [[0 1]]
2020-09-12T13:49:00.382Z        DEBUG   advmgr  sector-storage/sched.go:430     SCHED try assign sqi:0 sector 86 to window 0
2020-09-12T13:49:00.382Z        DEBUG   advmgr  sector-storage/sched.go:437     SCHED ASSIGNED sqi:0 sector 86 task seal/v0/addpiece to window 0
2020-09-12T13:49:00.382Z        DEBUG   advmgr  sector-storage/sched_resources.go:60    sched: not scheduling on worker 0 for startPreparing; not enough physical memory - need: 133285M, have 128851M
```

**A brief description of the problem you encountered while proving (sealing) a sector.**

Context: my current sector state is as follows:

```
79: PreCommit1 sSet: NO  active: NO  tktH: 50456 seedH: 0     deals: [0]
81: PreCommit1 sSet: NO  active: NO  tktH: 50456 seedH: 0     deals: [0]
82: PreCommit1 sSet: NO  active: NO  tktH: 0     seedH: 0     deals: [0]
84: PreCommit1 sSet: NO  active: NO  tktH: 0     seedH: 0     deals: [118421 119146 0 0 0 0 0 0 0 0 0 0 119509 119842 120772 0 0 0 0 0 0 0]
85: WaitDeals  sSet: NO  active: NO  tktH: 0     seedH: 0     deals: [121202 0 0 0 0 0 0 0 0 0 0 121468 121775 123012 123937 124778]
86: WaitDeals  sSet: NO  active: NO  tktH: 0     seedH: 0     deals: [122047]
87: WaitDeals  sSet: NO  active: NO  tktH: 0     seedH: 0     deals: [122284]
```

I have tried:

- Removing the sectors with no deals, which is ignored (both `sectors remove --really-do-it <N>` and `sectors update-state --really-do-it Removing` have zero effect on sectors any longer. My thoughts were, as those sectors with no deals were pre 0.7.0, that removing them might unblock my pipeline -- this was not the case.
- Rebuilding all binaries from scratch with a clean on all 3 machines, `--version` reports on all three: `lotus-miner version 0.7.0+git.232cc320`
- Rebooting the machines, and `lotus-daemon` itself.
- At the suggestion of @ribasushi , that perhaps the indices of the miners need to be sequential, or matter in some other way (I.E it is seeing worker 0 but not worker 2, 3 because it errors on 1), and thus restarting all my miners in order to make sure they get 0, 1, 2 sequential indices. Didn't help.

**Sectors status**

```
79: PreCommit1 sSet: NO  active: NO  tktH: 50456 seedH: 0     deals: [0]
81: PreCommit1 sSet: NO  active: NO  tktH: 50456 seedH: 0     deals: [0]
82: PreCommit1 sSet: NO  active: NO  tktH: 0     seedH: 0     deals: [0]
84: PreCommit1 sSet: NO  active: NO  tktH: 53019 seedH: 0     deals: [118421 119146 0 0 0 0 0 0 0 0 0 0 119509 119842 120772 0 0 0 0 0 0 0]
85: WaitDeals  sSet: NO  active: NO  tktH: 0     seedH: 0     deals: [121202 0 0 0 0 0 0 0 0 0 0 121468 121775 123012 123937 124778 125264]
86: WaitDeals  sSet: NO  active: NO  tktH: 0     seedH: 0     deals: [122047]
87: WaitDeals  sSet: NO  active: NO  tktH: 0     seedH: 0     deals: [122284]
```

**Lotus miner logs**

Lots of scheduling logs, but none which actually assign anything to anything, for example:

```
2020-09-12T22:43:31.214Z        DEBUG   advmgr  sector-storage/sched.go:324     SCHED 1 queued; 6 open windows
2020-09-12T22:43:31.217Z        DEBUG   advmgr  sector-storage/sched.go:415     SCHED windows: [{allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>}
 todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} tod
o:[]} {allocated:{memUsedMin:0 memUsedMax:0 gpuUsed:false cpuUse:0 cond:<nil>} todo:[]}]
2020-09-12T22:43:31.217Z        DEBUG   advmgr  sector-storage/sched.go:416     SCHED Acceptable win: [[]]
```

When it DOES try and assign something, it always chooses worker 0 and complains about not enough memory, the example near the beginning of this ticket shows it but here's the relevant line:

```
2020-09-12T13:49:00.382Z        DEBUG   advmgr  sector-storage/sched_resources.go:60    sched: not scheduling on worker 0 for startPreparing; not enough physical memory - need: 133285M, have 128851M
```

**Lotus miner diagnostic info**

Diagnostic information attached to the ticket.

**Code modifications**

No modifications

**Version**

`0.7.0+git.232cc320` for all workers, miner, and daemon.

**Comment by Reisen** [comment]
For anyone else who runs into this, I don't have a real fix, but I did manage to find a way out -- though it took 3 days. I had to remove all sectors. A problem at the moment which seems to be new, or MAY be related to my miner alone, is removing a sector is no longer immediate. I ended up issuing Removing several times, and it only took effect once a miner finished trying to process a sector for that stage.

Essentially, I had to wait for my main miner to process 5 PC1 stages, and throw them all away, completely sequentially. Once this was done, it started respecting my setup again and scheduling work over all miners.

Something about the sectors queued up in particular was causing scheduling issues, though I can't guess why.

**Comment by kelvinzhong** [comment]
me too, all jobs are assign to worker0, since last week and still not working....

**Comment by Reiers** [comment]
Hi @Reisen 

Thanks for the issue ticket. The issue is outdated, and Im not able to reproduce the issue on the latest tag.
If this is a issue in [current release](https://github.com/filecoin-project/lotus/releases/tag/v1.14.4) please open a new ticket - we do have an open ticket for something similar here: https://github.com/filecoin-project/lotus/issues/7868, but its not exactly the same.
Closing this one, and thanks again!
